src/contrib/profiling-sssd.rst

```diff
@@ -0,0 +1,108 @@
+Profiling SSSD
+##############
+
+Getting Ready
+*************
+
+There are several tools allowing us to profile an application. We will focus on
+perf_ which seems to be the one that works best with SSSD.
+
+.. _perf: https://perf.wiki.kernel.org
+
+First you need to install the ``perf`` tool:
+
+.. code-tabs::
+
+    .. fedora-tab::
+
+        # dnf -y install perf js-d3-flame-graph
+
+    .. ubuntu-tab::
+
+        $ sudo apt install linux-tools-common linux-tools-generic
+
+
+Make sure SSSD's and its dependencies' debug information is available, either
+by using ``debuginfod``, or by manually installing the debug information packages:
+
+.. code-tabs::
+
+    .. fedora-tab::
+
+        # dnf -y --enable-repo=fedora-debuginfo --enable-repo=updates-debuginfo \
+        install sssd*debuginfo libsss_*-debuginfo samba-client-libs-debuginfo \
+        libldb-debuginfo libtevent-debuginfo libtalloc-debuginfo glibc-debuginfo
+
+    .. ubuntu-tab::
+
+        $ sudo apt install sssd-ad-common-dbgsym sssd-ad-dbgsym sssd-common-dbgsym \
+        sssd-dbus-dbgsym sssd-ipa-dbgsym sssd-kcm-dbgsym sssd-krb5-common-dbgsym \
+        sssd-krb5-dbgsym sssd-ldap-dbgsym sssd-proxy-dbgsym sssd-tools-dbgsym \
+        libsss-certmap0-dbgsym libsss-idmap0-dbgsym libsss-nss-idmap0-dbgsym \
+        libsss-simpleifp0-dbgsym libsss-sudo-dbgsym libldb2-dbgsym libtdb1-dbgsym \
+        libtalloc2-dbgsym libtevent0-dbgsym libc6-dbgsym
+
+.. seealso::
+
+    More information on `Debug Symbol Packages`_ for Ubuntu.
+
+.. _`Debug Symbol Packages`: https://documentation.ubuntu.com/server/reference/debugging/debug-symbol-packages/
+
+.. note::
+
+    It is possible that SELinux prevents ``perf`` from monitoring your process.
+    If that happens, you can create the required rules or temporarily disable
+    selinux:
+
+    .. code-block:: console
+
+        # setenforce Permissive
+
+
+Profiling
+*********
+
+The simplest way to profile one of the SSSD's daemons is to attach the profiler
+to the process while it is running.
+
+Once SSSD is running and ready to be profiled, identify the PID of the process
+you want to monitor (``sssd_ssh``, ``sssd_nss``, ``sssd_be``, etc.), start the
```

Comment:
It would be nice to provide more examples, as in what data will be profiled selected the common processes. We will make the assumption that administrators may fully understand the SSSD architecture.

Reply:
Added one more example.